### ISSUE TITLE
Volumetric clouds not showing on certain worlds

### ISSUE THREAD

**Issue by ASaull** [X0nk/Bliss-Shader#145]
Using the latest git version, but this issue is not present in Bliss 2.0.4

Some worlds in my multiverse server do not show the volumetric clouds.

Things I've noticed
 - These worlds were generated with an old <1.3? version of minecraft
 - This also occurs for these worlds in singleplayer
 - The issue seems to be isolated to the region files, since copying these into a new world causes the issue for that world as well

**Comment by tr7zw** [comment]
I thought I had some config or mod issue, messing with settings for 10 mins... On the server, there are just no clouds for some reason. So can confirm this still happens on the current git version.

**Comment by AconexOfficial** [comment]
Yep, same problem for me. No clouds on my SMP world, but clouds on other some singleplayer worlds.

**Comment by X0nk** [comment]
this is extremely weird.
could anyone provided specific details? like mod setups / pc setups

**Comment by AconexOfficial** [comment]
> this is extremely weird. could anyone provided specific details? like mod setups / pc setups

I'm on 1.20.1 I have tried removing my clientside mods, but this also happens when I only use these 5 mods:
QuiltAPI 7.5.0
Sodium 0.5.8
Lithium 0.11.2
Indium 1.0.30
Iris 1.7.0

Also I sent my friends the exact same modpack I use
I haven't tried different mod versions of those though.
All I know is that I downloaded this shader on may 22nd and that version still has clouds on that SMP world.

PC is Windows 11, RTX 4070, i7-12700F

PC of a friend where I know that they have clouds on the SMP world uses RX7600, R5 7600X

**Comment by ASaull** [comment]
I just tested a clean Minecraft 1.20.4 profile with just Sodium 0.5.8 and Iris 1.7.0 installed, and I still have the issue.

I'm on Windows 11, RTX 3070, Ryzen 5600X

**Comment by ASaull** [comment]
OK, I looked at this again, and figured out the issue. The `worldDay` integer seems to be hitting its maximum value of 65535 whenever the DayTime value in level.dat reaches 65535 * 24000 = 1572840000.

This only happened on my worlds because before the doDaylightCycle gamerule was added I had them continuously resetting their time to noon, which actually jumps DayTime forward by 24000. For a normal world, hitting this limit would take 2.5 years of continuous running, which I imagine could happen on a server.

In Bliss 2.0.4 `worldDay` is only used for seasons, and I can confirm that seasons do also break on 2.0.4 on these worlds.

I'm not super familiar with GLSL, but my proposed solution is to just take only the first 16 bits of `worldDay` by replacing every instance of `worldDay` with `worldDay & 0xFFFF`. This seems to work for me. #

**Comment by Pumpk1in** [comment]
> I'm not super familiar with GLSL, but my proposed solution is to just take only the first 16 bits of `worldDay` by replacing every instance of `worldDay` with `worldDay & 0xFFFF`. This seems to work for me.

THANK YOU.
Over a year and your fix proposal is still not implemented. I don't get why, but thank you again.


For other people passing by, you can fix it if you edit the files in the shader .zip
You need to apply this patch (you can ignore climate_settings.glxl edits as he made a little mistake) :
- https://github.com/X0nk/Bliss-Shader/pull/299/commits/ff13358df09aa7c54fa76305f9859032ac8d269a

For climate_settings.glxl, apply this one instead :
- https://github.com/X0nk/Bliss-Shader/pull/299/commits/1dce580891a47e896b53fbef090ee62c155314bf
